spec/features/course/achievement_management_spec.rb

```diff
@@ -108,6 +108,158 @@
+            find('#condition_achievement_achievement_id').
+              find(:css, "option[value='#{achievement.id}']").
+              select_option
+            click_button 'helpers.submit.condition_achievement.create'
+          end
+          it 'creates an achievement condition correctly' do
+            expect(current_path).to eq edit_course_achievement_path(course, achievement)
+            expect(page).to have_selector('tr.condition > td:nth-child(2)', text: achievement.title)
+          end
+        end
+        context 'editing and deleting' do
+          let!(:achievement_which_is_condition) { create(:course_achievement, course: course) }
+          let(:achievement_condition) { create(:course_condition_achievement, course: course,
+                                               achievement: achievement_which_is_condition) }
+          let!(:some_other_achievement) { create(:course_achievement, course: course) }
+          let!(:achievement) { create(:course_achievement, course: course,
+                                      conditions: [achievement_condition]) }
+
+          it 'editing an achievement condition' do
+            expect(current_path).to eq edit_course_achievement_path(course, achievement)
+            expect(page).to have_selector('tr.condition > td:nth-child(2)',
+                                          text: achievement_which_is_condition.title)
+            condition_edit_path =
+              edit_course_achievement_condition_achievement_path(course,
+                                                                 achievement,
+                                                                 achievement_condition)
+            find_link(nil, href: condition_edit_path).click
+
+            find('#condition_achievement_achievement_id').
+              find(:css, "option[value='#{some_other_achievement.id}']").
+              select_option
+            click_button 'helpers.submit.condition_achievement.update'
+            expect(current_path).to eq edit_course_achievement_path(course, achievement)
+            expect(page).to have_selector('tr.condition > td:nth-child(2)',
+                                          text: some_other_achievement.title)
+          end
+
+          it 'deleting an achievement condition' do
+            condition_delete_path =
+              course_achievement_condition_achievement_path(course, achievement,
+                                                            achievement_condition)
+            expect do
+              find_link(nil, href: condition_delete_path).click
+            end.to change { achievement.conditions.count }.by(-1)
+            expect(page).not_to have_selector('tr.condition > td:nth-child(2)',
+                                              text: achievement_which_is_condition.title)
+          end
+        end
+      end
+
+      context 'achievement level conditions' do
+        it { is_expected.to have_selector(:link_or_button,
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.